bkbit/scripts/gfftranslator.py

```diff
@@ -1,203 +1,682 @@
+            otherwise None.
+
+        Raises:
+            None
+
+        """
+        stable_id = None
+        if "Dbxref" in attributes:
+            dbxref = {t.strip() for s in attributes["Dbxref"] for t in s.split(",")}
+            geneid_values = set()
+            for reference in dbxref:
+                k, v = reference.split(":", 1)
+                if k == "GeneID":
+                    geneid_values.add(v.split(".")[0])
+            if len(geneid_values) == 1:
+                stable_id = geneid_values.pop()
+        else:
+            logger.error(
+                "Line %s: No GeneAnnotation object created for this row due to missing dbxref attribute.",
+                curr_line_num,
+            )
+            return None
+
+        if not stable_id:
+            logger.error(
+                "Line %s: No GeneAnnotation object created for this row due to number of GeneIDs provided in dbxref attribute is not equal to one.",
+                curr_line_num,
+            )
+            return None
+
+        # Check and validate the name attribute
+        name = self.__get_attribute(attributes, "Name", curr_line_num)
+
+        # Check and validate the description attribute
+        description = self.__get_attribute(attributes, "description", curr_line_num)
+
+        # Check and validate the biotype attribute
+        biotype = self.__get_attribute(attributes, "gene_biotype", curr_line_num)
+
+        # Parse synonyms
+        synonyms = []
+        if "gene_synonym" in attributes:
+            synonyms = list(
+                {t.strip() for s in attributes["gene_synonym"] for t in s.split(",")}
+            )
+            synonyms.sort()  # note: this is not required, but it makes the output more predictable therefore easier to test
+        else:
+            logger.warning(
+                "Line %s: synonym is not set for this row's GeneAnnotation object due to missing gene_synonym attribute.",
+                curr_line_num,
+            )
+
+        gene_annotation = kbmodel.GeneAnnotation(
+            id=NCBI_GENE_ID_PREFIX + ":" + stable_id,
+            source_id=stable_id,
+            symbol=name,
+            name=name,
+            description=description,
+            molecular_type=biotype,
+            referenced_in=self.genome_annotation.id,
+            in_taxon=[self.organism_taxon.id],
+            in_taxon_label=self.organism_taxon.full_name,
+            synonym=synonyms,
+        )
+        if gene_annotation.id in self.gene_annotations:
+            if gene_annotation != self.gene_annotations[gene_annotation.id]:
+                return self.__resolve_ncbi_gene_annotation(
+                    gene_annotation, curr_line_num
+                )
+            if name != self.gene_annotations[gene_annotation.id].name:
+                logger.warning(
+                    "Line %s: GeneAnnotation object with id %s already exists with a different name.",
```

Comment:
Please add the conflicting names to the warning message

Reply:
fixed